src/main/java/com/faforever/client/chat/ChatUserService.java

```diff
@@ -42,9 +57,44 @@
   private final I18n i18n;
   private final EventBus eventBus;
 
+  private List<EmoticonsGroup> emoticonsGroups;
+  private HashMap<String, String> emoticonShortcodes;
+
   @Override
   public void afterPropertiesSet() {
     eventBus.register(this);
+    loadAndVerifyEmoticons();
+  }
+
+  private void loadAndVerifyEmoticons() {
+    try {
+      emoticonsGroups = new ObjectMapper().readValue(EMOTICONS_JSON_FILE_RESOURCE.getFile(),
+          TypeFactory.defaultInstance().constructCollectionType(List.class, EmoticonsGroup.class));
```

Comment:
We don't want to create a new ObjectMapper every time as they are expensive to create. We have an ObjectMapper that is created on application start and can be injected by adding it to the constructor through spring

Reply:
I also would just make a class that is a list of emoticons rather than using the type factory to deserialize